crud/common/utils.lua

```diff
@@ -148,6 +152,65 @@ function utils.extract_key(tuple, key_parts)
     return key
 end
 
+local keydef_cache = {}
+setmetatable(keydef_cache, {__mode = 'k'})
+
+local function new_light_keydef(index_parts)
+    local key = msgpack.encode({utils.get_index_paths(index_parts)})
+
+    if keydef_cache[key] ~= nil then
+        return keydef_cache[key]
+    end
+
+    local key_def = keydef_lib.new(index_parts)
+    keydef_cache[key] = key_def
+
+    return key_def
+end
```

Comment:
It starts to look stranger and stranger. Is msgpack.encode with utils.get_index_paths really faster than key_def.new?